### PR TITLE
Initial implementation to support Github Enterprise

### PR DESCRIPTION
**Pull request by amrnt**
Supporting `WTF_GITHUB_BASE_URL` and  `WTF_GITHUB_UPLOAD_URL`. 

- If `WTF_GITHUB_BASE_URL` wasn't provided it will talk to the "normal" Github
- If `WTF_GITHUB_UPLOAD_URL` wasn't provided it will get the same value as `WTF_GITHUB_BASE_URL`

### CODE REVIEW

File: github/github_repo.go [hunk 1]
```diff
@@ -66,6 +71,19 @@ func (repo *GithubRepo) oauthClient() *http.Client {
 	return oauth2.NewClient(context.Background(), tokenService)
 }
 
+func (repo *GithubRepo) githubClient() (*ghb.Client, error) {
+	oauthClient := repo.oauthClient()
+
+	if len(repo.baseURL) > 0 {
+		if len(repo.uploadURL) == 0 {
+			repo.uploadURL = repo.baseURL
```
Comment (senorprogrammer) [inline github/github_repo.go]: Where is `uploadURL` used? In the next line, `baseURL` is in there twice. I assume one should be `uploadURL`?

File: github/github_repo.go [hunk 2]
```diff
@@ -66,6 +71,19 @@ func (repo *GithubRepo) oauthClient() *http.Client {
...
+		}
```
Comment (senorprogrammer) [inline github/github_repo.go]: Would love to see all this condensed into a single function, something like `isGitHubEnterprise()`:
```
if repo.isGitHubEnterprise() {
  return ghb.NewEnterpriseClient(repo.baseURL, repo.baseURL, oauthClient)
}
return ghb.NewClient(oauthClient), nil
```

File: _site/content/posts/modules/github.md [hunk 3]
```diff
@@ -26,7 +26,13 @@ wtf/github/
...
 
 <span class="caption">Key:</span> `WTF_GITHUB_TOKEN` <br />
-<span class="caption">Action:</span> Your <a href="https://developer.github.com/v3/oauth_authorizations/#create-a-new-authorization">Github API</a> token.
+<span class="caption">Action:</span> Your <a href="https://developer.github.com/v3/oauth_authorizations/#create-a-new-authorization">Github API</a> token or <a href="https://developer.github.com/enterprise/2.13/v3/enterprise-admin/">Github Enterprise</a> API token.
+
+<span class="caption">Key:</span> `WTF_GITHUB_BASE_URL` <br />
+<span class="caption">Action:</span> Your <a href="https://developer.github.com/enterprise/2.13/v3/enterprise-admin/">Github Enterprise</a> API URL.
+
+<span class="caption">Key:</span> `WTF_GITHUB_UPLOAD_URL` <br />
+<span class="caption">Action:</span> Your <a href="https://developer.github.com/enterprise/2.13/v3/enterprise-admin/">Github Enterprise</a> upload URL (often the same as API URL).
```
Comment (senorprogrammer) [inline _site/content/posts/modules/github.md]: The documentation needs to be very clear about which ENV vars are required for connecting to GitHub, and which are required for connecting to GitHub Enterprise. I don't think they should be commingled in the docs.

Something like 

```
## GitHub Required ENV Variables

...

## GitHub Enterprise Required ENV Variables

...

```

Comment (amrnt) [reply _site/content/posts/modules/github.md]: I'm not sure about fixing the docs. Needs more clearance. Maybe setting a new prefix for GE, like `WTF_GE_TOKEN`?

Comment (senorprogrammer) [reply _site/content/posts/modules/github.md]: My main concern is that there be no confusion by anyone who's not a GitHub Enterprise user about what configuration they need to set up. They're likely to outnumber GHE users by a wide margin.

**Comment by senorprogrammer** [comment]
Great work, thanks for this!

**Pull request merged**